pyro/distributions/torch_distribution.py

```diff
@@ -243,9 +255,17 @@ def rsample(self, sample_shape=torch.Size()):
         return self.base_dist.rsample(sample_shape)
 
     def log_prob(self, value):
+        if self._mask is False:
+            shape = broadcast_shape(self.base_dist.batch_shape,
+                                    value.shape[:value.dim() - self.event_dim])
+            return torch.zeros((), device=value.device).expand(shape)
```

Comment:
You alluded to this in the description I think (but I didn't quite follow) - why can't we use the same `dtype` as `value` in the call to `torch.zeros`?

Reply:
Value could be long, but log_prob must be float of various precision.

Reply:
I suppose we could still disambiguate float32/64 and use the default tensor type for long, but not sure if its worth it.